src/corona_hakab_model/policies_manager.py

```diff
@@ -43,7 +48,18 @@ def activate_partial_opening_policies(self):
             # going through each policy activator.
             for conditioned_policy in conditioned_policies:
                 # check if temp is satisfied
-                self.update_matrix_manager.check_and_apply(con_type, circles, conditioned_policy, manager=self.manager)
+                activating_policy, affected_circles = self.update_matrix_manager.check_and_apply(
+                    con_type, circles, conditioned_policy, manager=self.manager)
+                # Append affected circles to the daily affected circles
+                if activating_policy:
+                    self.update_daily_affected_circles(affected_circles, conditioned_policy)
+
+    def update_daily_affected_circles(self, affected_circles, conditioned_policy):
+        if self.daily_affected_circles is None:
+            self.daily_affected_circles = {}
+        if conditioned_policy.message not in self.daily_affected_circles:
+            self.daily_affected_circles[conditioned_policy.message] = []
+        self.daily_affected_circles[conditioned_policy.message] += affected_circles
```

Comment:
Why you are using the conditioned_policy message as key?

Reply:
I replaced the dictionary's key to be a conditioned policy (instead of its message)